packages/core/src/util/json.ts

```diff
@@ -0,0 +1,78 @@
+const jsonify = (almostJson: string) => {
+  try {
+    return JSON.parse(almostJson)
+  } catch (e) {
+    // eslint-disable-next-line no-useless-escape
+    const tryFindJson = almostJson.replace(/([a-zA-Z0-9_$]+\s*):/g, '"$1":').replace(/'([^']+?)'([\s,\]\}])/g, '"$1"$2')
+    return JSON.parse(tryFindJson)
+  }
+}
+
+const chars = {
+  '[': ']',
+  '{': '}',
+}
+
+type StrIterator = (char: string, index: number, all: string) => undefined | boolean
+
+const any = (iteree: string, iterator: StrIterator) => {
+  let result
+  for (let i = 0; i < iteree.length; i++) {
+    result = iterator(iteree[i], i, iteree)
+    if (result) {
+      break
+    }
+  }
+  return result
+}
+
+const extract = (str: string) => {
+  // eslint-disable-next-line no-useless-escape
+  const startIndexRgex = /[\{\[]/
+  const startIndex = str.search(startIndexRgex)
+  if (startIndex === -1) {
+    return null
+  }
+
+  const openingChar = str[startIndex] as keyof typeof chars
+  const closingChar = chars[openingChar]
+  let endIndex = -1
+  let count = 0
+
+  str = str.substring(startIndex)
+  any(str, (letter, i) => {
+    if (letter === openingChar) {
+      count++
+    } else if (letter === closingChar) {
+      count--
+    }
+
+    if (!count) {
+      endIndex = i
+      return true
+    }
+  })
+
+  if (endIndex === -1) {
+    return null
+  }
+
+  return str.substring(0, endIndex + 1)
+}
+
+export function extractJsonFromString(str: string) {
+  let nStr = str
```

Comment:
I can't get the usage for `extractJsonFromString`, could you add some test case for it?